community/kernel/src/main/java/org/neo4j/kernel/impl/index/schema/NativeSchemaNumberIndexReader.java

```diff
@@ -0,0 +1,175 @@
+        FullScanNonUniqueIndexSampler<KEY,VALUE> sampler =
+                new FullScanNonUniqueIndexSampler<>( tree, layout, indexSamplingConfig );
+        return sampler::result;
+    }
+
+    @Override
+    public long countIndexedNodes( long nodeId, Value... propertyValues )
+    {
+        treeKeyFrom.from( nodeId, propertyValues );
+        treeKeyTo.from( nodeId, propertyValues );
+        try ( RawCursor<Hit<KEY,VALUE>,IOException> seeker = tree.seek( treeKeyFrom, treeKeyTo ) )
+        {
+            long count = 0;
+            while ( seeker.next() )
+            {
+                if ( seeker.get().key().entityId == nodeId )
+                {
+                    count++;
+                }
+            }
+            return count;
+        }
+        catch ( IOException e )
+        {
+            throw new UncheckedIOException( e );
+        }
+    }
+
+    @Override
+    public PrimitiveLongIterator query( IndexQuery... predicates ) throws IndexNotApplicableKernelException
+    {
+        if ( predicates.length != 1 )
+        {
+            throw new UnsupportedOperationException();
+        }
+
+        IndexQuery predicate = predicates[0];
+        switch ( predicate.type() )
+        {
+        case exists:
+            treeKeyFrom.initAsLowest();
+            treeKeyTo.initAsHighest();
+            return startSeekForInitializedRange();
+        case exact:
+            ExactPredicate exactPredicate = (ExactPredicate) predicate;
+            Value[] values = new Value[] {exactPredicate.value()};
+            treeKeyFrom.from( Long.MIN_VALUE, values );
+            treeKeyTo.from( Long.MAX_VALUE, values );
+            return startSeekForInitializedRange();
+        case rangeNumeric:
+            // todo: NumberRangePredicate should return NumberValue instead of Number
+            NumberRangePredicate rangePredicate = (NumberRangePredicate) predicate;
+            treeKeyFrom.from( rangePredicate.fromInclusive() ? Long.MIN_VALUE : Long.MAX_VALUE,
+                    new Value[] {Values.of( rangePredicate.from() )} );
+            treeKeyFrom.entityIdIsSpecialTieBreaker = true;
+            treeKeyTo.from( rangePredicate.toInclusive() ? Long.MAX_VALUE : Long.MIN_VALUE,
+                    new Value[] {Values.of( rangePredicate.to() )} );
+            treeKeyTo.entityIdIsSpecialTieBreaker = true;
+            return startSeekForInitializedRange();
+        default:
+            throw new IllegalArgumentException( "IndexQuery of type " + predicate.type() + " is not supported." );
+        }
+    }
+
+    @Override
+    public boolean hasFullNumberPrecision()
+    {
+        return true;
+    }
+
+    private PrimitiveLongIterator startSeekForInitializedRange()
+    {
+        try
+        {
+            RawCursor<Hit<KEY,VALUE>,IOException> seeker = tree.seek( treeKeyFrom, treeKeyTo );
+            openSeekers.add( seeker );
+            return new NumberHitIterator<>( seeker, openSeekers );
```

Comment:
That change would need to travel all the way up the stack for it to take any effect, right?